tests/unit/test_emulators.py

```diff
@@ -834,6 +834,45 @@ def test_predict_away_training_data_points(self):
             assert x not in training_inputs
             self.assertTrue(emulator.predict(x).variance > 0)
 
+    def test_small_training_data_warning(self):
+        """Ensure a correct Warning is raised if the size of the training set is less than the number of
+        hyperparameters in the Gaussian Process."""
+
+        emulator = MogpEmulator()
+
+        training_data = [TrainingDatum(Input(0), 1)]
+
+        with self.assertWarnsRegex(
+            UserWarning,
+            exact(
+                "Fewer training points (1) than hyperparameters (2) being "
+                "estimated. Estimates may be unreliable."
+            ),
+        ):
+            _ = emulator.fit(training_data)
```

Comment:
very minor, but `MogpEmulator.fit` doesn't return anything so should be:
```suggestion
            emulator.fit(training_data)
```